src/radical/pilot/agent/rm/lsf.py

```diff
@@ -48,26 +48,29 @@ def _configure(self):
         #
         lsf_nodes = [line.strip() for line in open(lsf_hostfile)]
         self._log.info("Found LSB_DJOB_HOSTFILE %s. Expanded to: %s",
-                      lsf_hostfile, lsf_nodes)
+                       lsf_hostfile, lsf_nodes)
         lsf_node_list = list(set(lsf_nodes))
 
         # Grab the core (slot) count from the environment
         # Format: hostX N hostY N hostZ N
         lsf_cores_count_list = map(int, lsb_mcpu_hosts.split()[1::2])
-        lsf_core_counts      = list(set(lsf_cores_count_list))
-        lsf_cores_per_node   = min(lsf_core_counts)
-        lsf_gpus_per_node    = self._cfg.get('gpus_per_node', 0) # FIXME GPU
-        lfs_lfs_per_node     = {'path' : self._cfg.get('lfs_path_per_node', None),
-                                'size' : self._cfg.get('lfs_size_per_node', 0)
-                               }
+        lsf_core_counts = list(set(lsf_cores_count_list))
+        lsf_sockets_per_node = self._cfg.get('sockets_per_node', 1)
+        # For now, assume all sockets have an equal number of cores and gpus
+        lsf_cores_per_socket = min(lsf_core_counts)/lsf_sockets_per_node
+        lsf_gpus_per_socket = self._cfg.get(
+            'gpus_per_node', 0)/lsf_sockets_per_node  # FIXME GPU
+
+        lsf_lfs_per_node = {'path': self._cfg.get('lfs_path_per_node', None),
+                            'size': self._cfg.get('lfs_size_per_node', 0)
+                            }
 
         self._log.info("Found unique core counts: %s Using: %d",
-                      lsf_core_counts, lsf_cores_per_node)
+                       lsf_core_counts, lsf_cores_per_socket)
 
         # node names are unique, so can serve as node uids
-        self.node_list      = [[node, node] for node in lsf_node_list]
-        self.cores_per_node = lsf_cores_per_node
-        self.gpus_per_node  = lsf_gpus_per_node
-        self.lfs_per_node   = lfs_lfs_per_node
-
-
+        self.node_list = [[node, node] for node in lsf_node_list]
```

Comment:
Why is the `node` inserted twice in the `node_list`? I really do not understand the reason. If it is for Summit specifically, then why didn't you fix all other attributes as well?

Reply:
I think there should be a comment above that line: "# node names are unique, so can serve as node uids".